contracts/LimitOrderSwap.sol

```diff
@@ -111,6 +110,8 @@ contract LimitOrderSwap is ILimitOrderSwap, Ownable, TokenCollector, EIP712, Ree
 
         // unwrap extra WETH in order to pay for ETH taker token and profit
         uint256 wethBalance = weth.balanceOf(address(this));
+        // the if statement cannot be covered by tests
+        // because the WETH withdrawal amount is always greater than the balance this contract has
```

Comment:
> because the WETH withdrawal amount is always greater than the balance this contract has

Doesn't this means the balance cannot cover the amount being withdrawn hence should revert?

Reply:
How is `wethBalance > wethToPay` guaranteed? Can't I pass in orders with total weth taker amount (i.e., `wethToPay`) greater than the weth maker amount collected (i.e., `wethBalance`)?

Reply:
I think the if statement here is just to prevent calling the `withdraw` function with a zero amount, instead of guaranteeing that `wethBalance` is greater than `wethToPay`. This is because we can craft a case to make `wethBalance` equal to `wethToPay`. However, we cannot craft a case to make `wethToPay` greater than `wethBalance`, as `wethToPay` is accumulated by `(makingAmount * order.takerTokenAmount) / order.makerTokenAmount`, and `makingAmount` is bounded by `orderAvailableAmount`.

The if statement can't cover it here due to the internal `require` statement implemented in the `withdraw` function of WETH, which we can't trigger under the condition `wethBalance - wethToPay` when `wethBalance > wethToPay`.

Reply:
> However, we cannot craft a case to make `wethToPay` greater than `wethBalance`, as `wethToPay` is accumulated by `(makingAmount * order.takerTokenAmount) / order.makerTokenAmount`, and `makingAmount` is bounded by `orderAvailableAmount`.

`makingAmount` and `orderAvailableAmount` only affect what percentage of `order.takerTokenAmount`. It can be 50% or 100% whatever, but `order.takerTokenAmount` is a user passed in parameter so I don't see why I can't pass in a large enough `order.takerTokenAmount` that will result in `wethToPay > wethBalance`.

Reply:
Yeah, you're right. We could make `wethToPay > wethBalance` by passing a large `order.takerTokenAmount`. However, as I mentioned before, the if statement here is to avoid calling the `withdraw` function with a zero amount, rather than guaranteeing that `wethBalance` is always greater than `wethToPay`. Therefore, the if statement cannot be covered because we cannot satisfy the internal `require` statement in WETH.

Reply:
I see. I misunderstood your comment. Then maybe move the comment into the `if` code block and rephrase a bit:
```solidity
        if (wethBalance > wethToPay) {
            // this if code block cannot be fully covered because WETH withdraw will always succeed as we have checked that wethBalance > wethToPay
            unchecked {
                weth.withdraw(wethBalance - wethToPay);
            }
        }
```